src/Parsers/ParserAlterQuery.cpp

```diff
@@ -856,12 +856,12 @@ bool ParserAlterQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
 
     if (alter_object_type == ASTAlterQuery::AlterObjectType::DATABASE)
     {
-        if (!parseDatabase(pos, expected, query->database))
+        if (!parseDatabaseASTPtr(pos, expected, query->database))
             return false;
     }
     else
     {
-        if (!parseDatabaseAndTableName(pos, expected, query->database, query->table))
+        if (!parseDatabaseAndTableASTPtr(pos, expected, query->database, query->table))
```

Comment:
By the name of this function `parseDatabaseAndTableASTPtr` it must be parallel to `parseDatabaseAndTableName`, but it doesn't -- differs in parameter of ParserIdentifier (`with_query_paramter`). Also may be it will be better not to add a new function, but reuse `parseDatabaseAndTableName` by adding an argument `with_query_paramter`, and then setting the result to query via `setDatabase()` and `setTable()` methods?

Reply:
`parseDatabaseAndTableName` parses `database` and `table` as strings and fill string class fields. `setDatabase()` & `getDatabase()` also set string values to ASTIdentifier. The problem is that` ASTIdentifier` stores parametrs in `ASTQueryParameter` children. `parseDatabaseAndTableASTPtr` on the other hand directry saves ASTIdentifiers and because of that parameters are saved.